docs/src/pages/components/tree-view/BarTreeView.js

```diff
@@ -115,24 +93,23 @@ const CustomContent = React.forwardRef(function CustomContent(props, ref) {
   };
 
   return (
-    // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions
-    <div
-      className={clsx(className, classes.root, contentClasses.root, {
-        [contentClasses.expanded]: expanded,
-        [contentClasses.selected]: selected,
-        [contentClasses.focused]: focused,
-        [contentClasses.disabled]: disabled,
+    <CustomContentRoot
+      className={clsx(className, classes.root, {
+        'Mui-expanded': expanded,
+        'Mui-selected': selected,
+        'Mui-focused': focused,
+        'Mui-disabled': disabled,
```

Comment:
Interesting, for another example I remember that I used another form that perhaps should be discussed.

https://github.com/mui-org/material-ui/blob/58c7e708a419bc8895774e964b8ef00b55d045ac/docs/src/pages/components/cards/RecipeReviewCard.tsx#L23

Reply:
I think it is a bit of a different use-case, but I see your point. I would say that for the pseudo states, we should always rely on the pseudo classes, as we do with the core components, as sometimes it would be impossible to override some css rules if we don't bump the specificity when overriding.

For example, for the disabled state, the core components are defining the styles with:

```
'&.Mui-disabled': { ... },
```

If we don't bump the specificity in the overrides, we won't be able to override.